### ISSUE TITLE
BigQuery: TABLE parameters in table-valued functions cause parsing error

### ISSUE THREAD

**Issue by sebastien-irco** [sqlfluff/sqlfluff#7650]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

BigQuery supports [table parameters](https://docs.cloud.google.com/bigquery/docs/table-functions#table_parameters) in table-valued functions (TVFs), using the `TABLE<column TYPE, ...>` syntax. SQLFluff's BigQuery dialect does not recognize this syntax and raises a parsing error.

### Expected Behaviour

SQLFluff should parse `TABLE<...>` parameter declarations in `CREATE TABLE FUNCTION` statements without errors.

### Observed Behaviour

```
== [test_table_param.sql] FAIL
L:   5 | P:  34 |  PRS | Line 5, Position 34: Found unparsable section:
                       | 'analytics_sandbox.test_tvf(\n  input_tabl...'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
```

### How to reproduce

```sql
CREATE OR REPLACE TABLE FUNCTION analytics_sandbox.test_tvf(
  input_table TABLE<sales INT64, item STRING>,
  item_name STRING
) AS (
  SELECT
    SUM(sales) AS total_sales,
    item
  FROM input_table
  WHERE item = item_name
  GROUP BY item
);
```

```bash
$ sqlfluff lint --dialect bigquery test_table_param.sql
```

### Dialect

BigQuery

### Version

Tested on both 3.5.0 and 4.0.4 (latest as of March 2026). Same parsing error on both.

### Configuration

Default (only `dialect = bigquery` is relevant for reproduction).

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
